tests/integration/loading/test_loading_component.py

```diff
@@ -0,0 +1,268 @@
+from multiprocessing import Lock
+
+import dash
+from dash.dependencies import Input, Output
+import dash_core_components as dcc
+import dash_html_components as html
+
+
+def test_ldcp001_loading_component_initialization(dash_dcc):
+    lock = Lock()
+
+    app = dash.Dash(__name__)
+
+    app.layout = html.Div(
+        [dcc.Loading([html.Div(id="div-1")], className="loading")], id="root"
+    )
+
+    @app.callback(Output("div-1", "children"), [Input("root", "n_clicks")])
+    def updateDiv(children):
+        with lock:
```

Comment:
wow very nice 💯

Reply:
Only the last test here (006) is new, the others I just pulled out of [test_integration_1.py](https://github.com/plotly/dash-core-components/pull/740/files#diff-47574444ebc5cb78be0c75fa212cc77a) into the `dash.testing` format before fixing them for the `className` update

Reply:
@HammadTheOne, @harryturr When writing tests involving callbacks, using a `Lock` like ☝️ is the best practice. Gives you full control of when the callback and your test conditions will be run in respect to one another.